packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+import * as React from 'react';
+import * as shape from 'd3-shape';
+import { classNamesFunction, getRTL } from '@fluentui/react/lib/Utilities';
+import {
+  IGaugeChartProps,
+  IGaugeChartSegment,
+  IGaugeChartStyleProps,
+  IGaugeChartStyles,
+  GaugeValueFormat,
+} from './GaugeChart.types';
+import { IProcessedStyleSet } from '@fluentui/react/lib/Styling';
+import {
+  Points,
+  convertToLocaleString,
+  formatValueWithSIPrefix,
+  getAccessibleDataObject,
+  getColorFromToken,
+  pointTypes,
+} from '../../utilities/index';
+import { ILegend, LegendShape, Legends, Shape } from '../Legends/index';
+import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
+import { Callout, DirectionalHint } from '@fluentui/react/lib/Callout';
+import { IYValueHover } from '../../index';
+import { TooltipText } from '../../utilities/TooltipText';
+import { select as d3Select } from 'd3-selection';
+
+const GAUGE_MARGIN = 16;
+const LABEL_WIDTH = 36;
+const LABEL_HEIGHT = 16;
+const LABEL_OFFSET = 4;
+const TITLE_OFFSET = 11;
+const EXTRA_NEEDLE_LENGTH = 4;
+export const ARC_PADDING = 2;
+export const BREAKPOINTS = [52, 70, 88, 106, 124, 142];
+const ARC_WIDTHS = [12, 16, 20, 24, 28, 32];
+export const FONT_SIZES = [20, 24, 32, 32, 40, 40];
```

Comment:
Looks like these 3 arrays will always be used with the same index. Is it better to put them in a single object.

Reply:
done